providers/lib/download/http.go

```diff
@@ -38,21 +38,25 @@ func (e Err) Error() string {
 	return fmt.Sprintf("http error %s:\n %q", e.Status, e.Body)
 }
 
-// Query will query the given URL and then return the response if response status is HTTP.OK
-// if status is not HTTP.OK, returns Err
+// Get will call GetWithClient using the default client
 func Get(url string, header http.Header) (*http.Response, error) {
+	if client, err := Client(); err != nil {
+		return nil, fmt.Errorf("can't obtain default client: %v", err)
+	} else {
```

Comment:
I'd dropped `else` -- the clauses are not at the same level, the one above handles the error, the one below is the expected flow of the program.